bib/cmd/bootc-image-builder/image.go

```diff
@@ -162,30 +270,24 @@ func manifestForDiskImage(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest
 	if !ok {
 		return nil, fmt.Errorf("pipelines: no partition tables defined for %s", c.Architecture)
 	}
-	// XXX: extract into helper
+
```

Comment:
Maybe still worth extracting the partition generation into a helper? This would also help with testing via image_test.go as the whole partition handling could be tested directly from go in a "integrationish" fashion. 

Attached is a sketched out idea that extracts the helper and does a single test to showcase the idea. The test of course would need to be transformed into a table test but I think it's a nice way to ensure that partition tests are quick

[0001-image-extract-genPartitionTable-helper-and-add-unit-.patch.txt](https://github.com/user-attachments/files/16610050/0001-image-extract-genPartitionTable-helper-and-add-unit-.patch.txt)

Reply:
Added.  Thanks!